component/loki/source/file/file_test.go

```diff
@@ -76,6 +76,63 @@ func Test(t *testing.T) {
 	}
 }
 
+func TestFileWatch(t *testing.T) {
+	defer goleak.VerifyNone(t, goleak.IgnoreTopFunction("go.opencensus.io/stats/view.(*worker).start"))
+	ctx, cancel := context.WithCancel(componenttest.TestContext(t))
+
+	// Create file to log to.
+	f, err := os.CreateTemp(t.TempDir(), "example")
+	require.NoError(t, err)
+	defer f.Close()
+
+	ctrl, err := componenttest.NewControllerFromID(util.TestLogger(t), "loki.source.file")
+	require.NoError(t, err)
+
+	ch1 := loki.NewLogsReceiver()
+
+	args := Arguments{
+		Targets: []discovery.Target{{
+			"__path__": f.Name(),
+			"foo":      "bar",
+		}},
+		ForwardTo: []loki.LogsReceiver{ch1},
+		FileWatch: FileWatch{
+			MinPollFrequency: time.Millisecond * 500,
+			MaxPollFrequency: time.Millisecond * 500,
+		},
+	}
+
+	go func() {
+		err := ctrl.Run(ctx, args)
+		require.NoError(t, err)
+	}()
+
+	ctrl.WaitRunning(time.Minute)
+
+	timeBeforeWriting := time.Now()
+
+	// Sleep for 600ms to miss the first poll, the next poll should be 2*MinPollFrequency later.
+	time.Sleep(time.Millisecond * 600)
+
+	_, err = f.Write([]byte("writing some text\n"))
+	require.NoError(t, err)
+
+	select {
+	case logEntry := <-ch1.Chan():
+		require.Greater(t, time.Since(timeBeforeWriting), 1*time.Second)
```

Comment:
yeah if my logic is correct, by setting the `MinPollFrequency` to 500ms, we will miss the first tick by sleeping for 600ms. That means that the next tick will be 1 second later which in total is 1.5 second after the write.
If the value of `MinPollFrequency` was not set correctly, it would be at 250ms. That means that the first tick would be missed by sleeping for 600ms, so the next tick would be 500ms later which in total is 750ms after the write. 750ms < 1 second, so the test would fail